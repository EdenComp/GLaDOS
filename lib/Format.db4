fn println(str) {
    print(str + "\n");
}

fn btoi(b) {
    if (b) {
        return 1;
    } else {
        return 0;
    }
}

fn itob(i) {
    if (i == 0) {
        return false;
    } else {
        return true;
    }
}
