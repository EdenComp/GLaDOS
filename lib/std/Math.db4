// Package std -- Math

fn is_even(a) {
    return (a % 2) == 0;
}

fn is_neg(a) {
    return a < 0;
}

fn abs(a) {
    if (a < 0) {
        return a * -1;
    }
    return a;
}

fn max(a, b) {
    if (a >= b) {
        return a;
    }
    return b;
}

fn min(a, b) {
    if (a <= b) {
        return a;
    }
    return b;
}

fn clamp(nb, min_nb, max_nb) {
    int low = min(nb, max_nb);

    return max(low, min_nb);
}

fn factorial(x) {
    if (x < 0) {
        return 0;
    } elif (x < 2) {
        return 1;
    } else {
        return x * factorial(x - 1);
    }
}
