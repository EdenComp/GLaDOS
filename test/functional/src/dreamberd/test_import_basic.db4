import "std:Format";

println("Import from standard is working!");
