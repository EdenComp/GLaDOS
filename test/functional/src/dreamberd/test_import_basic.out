Import from standard is working!
